### PR TITLE
FOGL-2657: improve css and dropdown appearance

### PR DESCRIPTION
**Pull request by mshariq-nerd**

### CODE REVIEW

File: src/app/components/core/scheduler/update-schedule/update-schedule.component.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
-import { Component, EventEmitter, Input, OnChanges, OnInit, Output, SimpleChanges } from '@angular/core';
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output, SimpleChanges, forwardRef } from '@angular/core';
```
Comment (monika-Dianomic) [inline src/app/components/core/scheduler/update-schedule/update-schedule.component.ts]: Is `forwardRef ` being used anywhere?

File: src/app/components/core/asset-readings/readings-graph/readings-graph.component.html [hunk 2]
```diff
@@ -2,29 +2,45 @@
...
-            <div class="select is-small">
-              <select #selectedTime (change)="getTimeBasedAssetReadingsAndSummary(selectedTime.value)">
-                <!-- <option value="0" [selected]="optedTime == 0">None</option> -->
-                <option value="600" [selected]="optedTime == 600">10 minutes</option>
-                <option value="1800" [selected]="optedTime == 1800">30 minutes</option>
-                <option value="3600" [selected]="optedTime == 3600">1 hour</option>
-                <option value="28800" [selected]="optedTime == 28800">8 hours</option>
-                <option value="86400" [selected]="optedTime == 86400">1 day</option>
-              </select>
+            <div id="time-dropdown" class="dropdown is-left">
```
Comment (praveen-garg) [inline src/app/components/core/asset-readings/readings-graph/readings-graph.component.html]: I think we should keep the same size, as in develop. And 
click the dropdown, dropdown appears correctly.
close and re-open the modal, the dropdown is open. On close it should have been reset.

File: src/app/components/core/audit-log/audit-log.component.html [hunk 3]
```diff
@@ -21,28 +21,59 @@ <h6 class='subtitle is-6'>
...
-            <div class="select is-fullwidth is-small">
-              <select (change)="filterSource('severity', $event)">
-                <option selected="selected">Severity</option>
-                <option *ngFor="let severity of logSeverityList" [value]='severity.name'>{{severity.name}}</option>
-              </select>
+            <div id="severity-dropdown" class="dropdown">
+              <div class="dropdown-trigger">
+                <button class="button is-fullwidth is-small" aria-haspopup="true" aria-controls="dropdown-menu"
+                  (click)="toggleDropDown('severity-dropdown')">
+                  <span>{{severity == ''? "Severity": severity.toUpperCase()}}</span>
```
Comment (praveen-garg) [inline src/app/components/core/audit-log/audit-log.component.html]: What is SUCCESS severity?

File: src/app/components/core/scheduler/update-schedule/update-schedule.component.html [hunk 4]
```diff
@@ -50,11 +51,24 @@
...
               <label class="label is-small">Type</label>
               <div class="control">
-                <div class="select is-fullwidth is-small">
-                  <select #selectedTypeValue (change)="selectedScheduleType = setScheduleTypeKey(selectedTypeValue.value)"
-                    formControlName="type">
-                    <option [selected]="type.name == type" *ngFor="let type of scheduleType">{{type.name}}</option>
-                  </select>
+                <div id="type-dropdown" class="dropdown">
+                  <div class="dropdown-trigger">
+                    <button class="button is-fullwidth is-small" aria-haspopup="true" aria-controls="dropdown-menu"
```
Comment (praveen-garg) [inline src/app/components/core/scheduler/update-schedule/update-schedule.component.html]: last item `Manual` is not visible

File: src/app/components/core/system-log/system-log.component.html [hunk 5]
```diff
@@ -21,27 +21,61 @@ <h6 class='subtitle is-6'>
...
-            <div class="select is-fullwidth is-small">
-              <select (change)="filterData($event, 'source')">
-                <option selected="selected" value="all">All</option>
-                <option value="storage">Storage</option>
-              </select>
+            <div id="dropdown" class="dropdown">
+              <div class="dropdown-trigger">
+                <button class="button is-fullwidth is-small" aria-haspopup="true" aria-controls="dropdown-menu"
+                  (click)="toggleDropDown('dropdown')">
+                  <span>{{source == ''? "All": source.toUpperCase()}}</span>
```
Comment (praveen-garg) [inline src/app/components/core/system-log/system-log.component.html]: Select Storage. Dropdown contents get hidden

File: src/app/components/core/system-log/system-log.component.html [hunk 6]
```diff
@@ -21,27 +21,61 @@ <h6 class='subtitle is-6'>
...
-              <select (change)="filterData($event, 'level')">
-                <option selected="selected" value="">Info</option>
-                <option value="WARNING">Warning</option>
-                <option value="ERROR">Error</option>
-              </select>
+            <div id="level-dropdown" class="dropdown">
+              <div class="dropdown-trigger">
+                <button class="button is-fullwidth is-small" aria-haspopup="true" aria-controls="dropdown-menu"
+                  (click)="toggleDropDown('level-dropdown')">
+                  <span>{{level == ''? "Info": level.toUpperCase()}}</span>
```
Comment (praveen-garg) [inline src/app/components/core/system-log/system-log.component.html]: all should be displayed in uppercase

File: src/app/components/core/settings/settings.component.html [hunk 7]
```diff
@@ -9,11 +9,22 @@
...
+                <span>{{protocol}}</span>
+                <span class="icon is-small">
+                  <i class="fa fa-angle-down" aria-hidden="true"></i>
+                </span>
+              </button>
+            </div>
+            <div class="dropdown-menu" id="dropdown-menu" role="menu">
+              <div class="dropdown-content">
+                <a class="dropdown-item" (click)="toggleDropDown('protocol-dropdown');setProtocol('http')">http</a>
+                <a class="dropdown-item" (click)="toggleDropDown('protocol-dropdown');setProtocol('https')">https</a>
```
Comment (praveen-garg) [inline src/app/components/core/settings/settings.component.html]: Not fully visible

File: src/app/components/core/configuration-manager/configuration-manager.component.html [hunk 8]
```diff
@@ -4,11 +4,25 @@
     <div class="column is-one-quarter">
       <nav class="panel">
         <div class="panel-heading">
-          <div class="select is-rounded is-fullwidth is-small">
-            <select (change)="getChildren($event.target.value)">
-              <option *ngFor="let rootCategory of rootCategories" [selected]='rootCategory.key === selectedRootCategory' [value]='rootCategory.key'> {{ hasProperty(rootCategory, 'displayName') === true ?
-                  rootCategory.displayName:rootCategory.description}}</option>
-            </select>
+          <div id="dropdown" class="dropdown is-vcentered">
```
Comment (monika-Dianomic) [inline src/app/components/core/configuration-manager/configuration-manager.component.html]: is-vcentered -> is-centered

File: src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts [hunk 9]
```diff
@@ -91,6 +91,10 @@ export class ReadingsGraphComponent implements OnDestroy {
     }
     this.isAlive = false;
     chart_modal.classList.remove('is-active');
+    const activeDropDowns = Array.prototype.slice.call(document.querySelectorAll('.dropdown.is-active'));
```
Comment (monika-Dianomic) [inline src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts]: It should be `activeDropDown`, not `activeDropDowns`.

**Pull request merged**